### ISSUE TITLE
Implementing a centralized repository for our GitHub Actions configurations

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#17856]
### ESLint version

HEAD

### What problem do you want to solve?

Currently, our main ESLint repository benefits from the automation provided by Renovate for dependency management. However, this practice has not been uniformly adopted across all our repositories. To address this, I propose the creation of a centralized repository within our ESLint organization. This repository will not only contain configurations for Renovate but also other common GitHub Actions workflows.

Advantages of this Approach:

* Unified Workflow Management: By centralizing our GitHub Actions configurations, including Renovate, we can ensure all ESLint projects follow a consistent workflow for automated tasks, thus improving our overall efficiency.

* Reduction in Redundant Efforts: A central repository eliminates the need to duplicate GitHub Actions configurations in each repository. This reduces repetitive work and ensures uniformity across our projects.

* Simplified Updates and Maintenance: Any changes or updates to our automation configurations would need to be made only once, in the central repository. This simplifies the process of maintaining and updating our workflows.

* Extensibility for ESLint Community Projects: This centralized repository can also be made available to projects under the `eslint-community`org, allowing them to benefit from standardized GitHub Actions workflows.

refs:
* https://github.com/eslint-community/eslint-plugin-n/pull/44
* https://github.com/eslint/eslint/discussions/17424

### What do you think is the correct solution?

Creation of a Central Repository: Establish a new repository, possibly named `eslint-gha-configs`, to house our GitHub Actions configurations:

* renovate
* release-please
* add-to-triage
* stale
* codeql

### Participation

- [x] I am willing to submit a pull request for this change.

### Additional comments

I'm sure Renovate config can be reused in this way, but I'm not sure if others (like `release-please`) are also feasible. help wanted!🙌

**Comment by kecrily** [comment]
This is definitely a good idea. No more repetitive work

**Comment by voxpelli** [comment]
Some prior art to this:

* [`fastify/workflows`](https://github.com/fastify/workflows) – reusable workflows for use in the Fastify organization 
* [`mdn/workflows`](https://github.com/mdn/workflows) – reusable GitHub Actions workflows for use in MDN repositories.
* [`SocketDev/workflows`](https://github.com/SocketDev/workflows) – which I created when I worked at Socket
* [`voxpelli/ghatemplates`](https://github.com/voxpelli/ghatemplates) – old name, but the place where I keep my personal ones

When it comes to renovate config, that's best stored in its own `eslint/renovate-config` repo, like eg [`voxpelli/renovate-config`](https://github.com/voxpelli/renovate-config), as then renovate automatically adds that config in new repos, see: https://docs.renovatebot.com/config-presets/#grouporganization-level-presets

**Comment by nzakas** [comment]
Sorry I missed this. This seems like a good idea to me. Let's get a few more eyes on this @eslint/eslint-team

**Comment by JoshuaKGoldberg** [comment]
This sounds very useful. Both for standardizing and for bringing the best of each repo's configurations to the rest of them.

It would be nice to have tooling to sync any changes from the template to downstream repositories - or vice-versa. It can easily happen that shared workflows fall out of sync when they have many downstream repos. Especially if this is to be used in both the `eslint` and `eslint-community` organizations.

https://github.com/JoshuaKGoldberg/create-typescript-app acts as a template repo with my personal workflows and a script to migrate existing repos to the workflows. I'm _hoping_ to split a general engine out of it later this year that could be used to power script syncing useful for this scenario.

**Comment by aladdin-add** [comment]
as a start, I've made a repo: https://github.com/aladdin-add/eslint-gha-configs. I will try to:

* Transfer it to eslint org, 
* Or create a new repo in eslint org and prepare a PR to it. (if you have a license concern)

**Comment by nzakas** [comment]
Yeah, we should just do a new repo from scratch to avoid any licensing issues. @mdjermanovic any thoughts before creating a `workflows` repo?

**Comment by nzakas** [comment]
Ping @mdjermanovic

**Comment by lumirlumir** [comment]
It looks like this suggestion aligns with the one I previously proposed: consolidating the `update-readme` script across repositories.

Here’s the related issue: https://github.com/eslint/rewrite/issues/200

Would it be alright to mark it as accepted, or should we have some more discussions?

**Comment by nzakas** [comment]
Let's get @eslint/eslint-tsc to chime in. Shall we move forward with this?

**Comment by fasttime** [comment]
> Shall we move forward with this?

No objections from my side. I think that the added cost of maintaining one more repo will be balanced by the benefit of the centralized custom actions and shared workflows.

**Comment by nzakas** [comment]
Okay, marking as accepted. What should the repo be named?

**Comment by lumirlumir** [comment]
I've found some real-world use cases. Here are some examples:

* Babel: [`actions`](https://github.com/babel/actions)
* Webpack: [`tooling`](https://github.com/webpack/tooling) – This repo isn't focused solely on GitHub Actions, but it's for tooling across Webpack repositories.

So, I think if we focus only on GitHub Actions, then `actions` would be a better fit; if it includes other tooling as well, then `tooling` might be more appropriate.

(FYI, I tried to find other repos like Vite, Unjs, Vercel, and others, but there were no specific repositories dedicated to GitHub Actions.)

**Comment by voxpelli** [comment]
I think `workflows` seems to be the most common name, see eg the links here: https://github.com/SocketDev/workflows#similar-projects

**Comment by nzakas** [comment]
https://github.com/eslint/workflows/

**Comment by aladdin-add** [comment]
1st pr to add renovate: https://github.com/eslint/workflows/pull/1. 🎉

**Comment by nzakas** [comment]
The repository has been created so we can continue working there.
